*** ./init.org	2010-06-04 12:05:56.156153980 -0700
--- ./init	2010-06-04 12:07:38.824904433 -0700
***************
*** 134,141 ****
    if [ -r /mnt-system/"$knoppix_dir"/KNOPPIX ]; then
     echo -e "\r${CRE}${GREEN}$DISTRO ${FOUNDAT}: ${MAGENTA}$dev${NORMAL}       "
     return 0
    fi
-   umount /mnt-system
   done
   return 1
  }
--- 134,154 ----
    if [ -r /mnt-system/"$knoppix_dir"/KNOPPIX ]; then
     echo -e "\r${CRE}${GREEN}$DISTRO ${FOUNDAT}: ${MAGENTA}$dev${NORMAL}       "
     return 0
+   elif [ -n "$ISOFILE" -a -r "/mnt-system/$ISOFILE" ]; then
+       umount /mnt-system
+       mkdir -p /mnt-basesystem
+       trymount "$dev" /mnt-basesystem >/dev/null 2>&1
+       mount -t iso9660 -o loop,ro "/mnt-basesystem/$ISOFILE" /mnt-system
+       if [ -r /mnt-system/"$knoppix_dir"/KNOPPIX ]; then
+ 	  echo -e "\r${CRE}${GREEN}$DISTRO ${FOUNDAT}: ${MAGENTA}${dev}/${ISOFILE}${NORMAL}       "
+ 	  return 0
+       else
+ 	  umount /mnt-system
+ 	  umount /mnt-basesystem
+       fi
+   else
+       umount /mnt-system
    fi
   done
   return 1
  }
***************
*** 283,293 ****
  
  if [ -n "$TSCLIENT" ]; then
   # Split cmdline, find NFSDIR variable
!  for i in $CMDLINE; do case "$i" in [Ii][Pp]=*|[Nn][Mm]=*|[Gg][Ww]=*|[Nn][Ff][Ss][Dd][Ii][Rr]=*) eval $i;; esac; done
   [ -n "$ip" ] && IP="$ip"
   [ -n "$nm" ] && NM="$nm"
   [ -n "$gw" ] && GW="$gw"
   [ -n "$nfsdir" ] && NFSDIR="$nfsdir"
  
   echo ""
   sleep 5
--- 296,307 ----
  
  if [ -n "$TSCLIENT" ]; then
   # Split cmdline, find NFSDIR variable
!  for i in $CMDLINE; do case "$i" in [Ii][Pp]=*|[Nn][Mm]=*|[Gg][Ww]=*|[Nn][Ff][Ss][Dd][Ii][Rr]=*|fromiso=*) eval $i;; esac; done
   [ -n "$ip" ] && IP="$ip"
   [ -n "$nm" ] && NM="$nm"
   [ -n "$gw" ] && GW="$gw"
   [ -n "$nfsdir" ] && NFSDIR="$nfsdir"
+  [ -n "$fromiso" ] && ISOFILE="$fromiso"
  
   echo ""
   sleep 5
