*** ./init.org	2010-05-26 10:39:28.618377576 -0700
--- ./init	2010-05-26 12:16:11.408357817 -0700
***************
*** 61,66 ****
--- 61,69 ----
  if grep -i -q "livecd_config=[a-zA-Z0-9/]" /proc/cmdline; then
  	export LIVECD_CONFIG=`sed 's/.*livecd_config=\([^ ]*\).*/\1/' /proc/cmdline`
  fi
+ if grep -i -q "isofile=[^ ]*" /proc/cmdline; then
+     export ISOFILE=$(sed 's/.*isofile=\([^ ]*\).*/\1/' /proc/cmdline)
+ fi
  
  #======================================
  # 3) Prepare module load support 
*** ./include.org	2010-05-26 11:08:03.408341783 -0700
--- ./include	2010-05-26 11:19:59.348394729 -0700
***************
*** 2159,2164 ****
--- 2159,2171 ----
  					#--------------------------------------
  					IFS=$IFS_ORIG
  					return
+ 				elif [ -n "$ISOFILE" -a -f /cdrom/$ISOFILE ]; then
+ 				    echo "Mounting ISO image.."
+ 				    umount $i > /dev/null
+ 				    mkdir -p /liveiso
+ 				    mount -v $cdopt -o ro $i /liveiso
+ 				    mount -v -t iso9660 -o loop,ro /liveiso/$ISOFILE /cdrom
+ 				    return
  				fi
  				umount $i &>/dev/null
  			done
*** ./linuxrc.org	2010-05-26 12:20:10.438374727 -0700
--- ./linuxrc	2010-05-26 12:20:47.508349174 -0700
***************
*** 61,66 ****
--- 61,69 ----
  if grep -i -q "livecd_config=[a-zA-Z0-9/]" /proc/cmdline; then
  	export LIVECD_CONFIG=`sed 's/.*livecd_config=\([^ ]*\).*/\1/' /proc/cmdline`
  fi
+ if grep -i -q "isofile=[^ ]*" /proc/cmdline; then
+     export ISOFILE=$(sed 's/.*isofile=\([^ ]*\).*/\1/' /proc/cmdline)
+ fi
  
  #======================================
  # 3) Prepare module load support 
