*** ./scripts/live.org	2010-07-01 13:23:10.365649565 -0700
--- ./scripts/live	2010-07-01 13:33:13.865625651 -0700
***************
*** 378,383 ****
--- 378,387 ----
  
  				LOCATION="/tmp/$(basename "${location}")"
  				;;
+ 			iso-scan/filename=*)
+ 			        ISOFILE="${ARGUMENT#iso-scan/filename=}"
+ 				export ISOFILE
+ 				;;
  
  			*/*=*)
  				question="${ARGUMENT%%=*}"
***************
*** 514,519 ****
--- 518,530 ----
  				return 0
  			fi
  		done
+ 	elif [ -n "$ISOFILE" -a -f "${DIRECTORY}/$ISOFILE" ]; then
+ 	    # remount ISO
+ 	    mkdir -p /isodevice > /dev/null 2>&1
+ 	    mount --move -n $DIRECTORY /isodevice > /dev/null 2>&1
+ 	    loopdev=$(setup_loop "/isodevice/$ISOFILE" "loop" "/sys/block/loop*" '0' '')
+ 	    mount -o loop -t iso9660 $loopdev $DIRECTORY >/dev/null 2>&1
+ 	    return 0
  	fi
  
  	return 1
