*** lib/svc/method/live-fs-root.org	Wed Jul 14 14:07:25 2010
--- lib/svc/method/live-fs-root	Wed Jul 14 16:39:38 2010
***************
*** 206,211 ****
--- 206,219 ----
  			then
  				echo "" > /.liveusb
  				break
+ 			elif [ -f /.cdrom/iso/osol-0906-x86.iso ]; then
+ 				/sbin/umount -f $dev
+ 				mkdir -p /isodevice
+ 				/sbin/mount -o nologging,noatime $dev /isodevice
+ 				LODEV=$(/usr/sbin/lofiadm -a /isodevice/iso/osol-0906-x86.iso)
+ 				/sbin/mount -o nologging,noatime $LODEV /.cdrom
+ 				echo "" > /.liveusb
+ 				break
  			else
  				/sbin/umount -f $dev
  				continue
