*** ./init.org	2010-07-08 16:48:13.000000000 -0400
--- ./init	2010-07-09 07:00:07.000000000 -0400
***************
*** 496,505 ****
   FND_SWAPFILEPUP="`find /mnt/data -maxdepth 2 -xdev -type f -name pupswap.swp | grep -v ' ' | sed -e 's%^/mnt/data%%g' | tr '\n' ' '`"
   #v2.22 improved folder name search, avoid false hits...
   FND_MULTIFOLDER="`find /mnt/data -maxdepth 1 -xdev -type d -name 20[0-9][0-9]-[0-9][0-9]-[0-9][0-9]-[0-9][0-9]-[0-9][0-9] | grep -v ' ' | sed -e 's%^/mnt/data%%g' | tr '\n' ' '`"
   [ "$FND_MULTIFOLDER" != "" ] && echo "$ONEDEV $ONEFS $FND_MULTIFOLDER" >> /tmp/MULTIFOLDERS
   #all written on one line...
   echo -n "$ONEDEV $ONEFS " >> /tmp/PUPPYFILES
!  echo "FND_QUIRKYKERNEL $FND_INITRDGZ $FND_PUPXXXSFS $FND_ZDRVSFS $FND_PUPSAVE2FS $FND_MULTIFOLDER $FND_FULLINSTALL $FND_SWAPFILEPUP" >> /tmp/PUPPYFILES
   [ -f /mnt/data/SAVEMARK ] && PSAVEMARK="`cat /mnt/data/SAVEMARK`" #v3.97 partition# that has or will-have ${DISTRO_FILE_PREFIX}save.
   #100318 fix any improper shutdowns... fsckme.err format: sda7,ext3,[/pupsave.2fs] (see rc.shutdown)
   FSCKME=""
--- 496,510 ----
   FND_SWAPFILEPUP="`find /mnt/data -maxdepth 2 -xdev -type f -name pupswap.swp | grep -v ' ' | sed -e 's%^/mnt/data%%g' | tr '\n' ' '`"
   #v2.22 improved folder name search, avoid false hits...
   FND_MULTIFOLDER="`find /mnt/data -maxdepth 1 -xdev -type d -name 20[0-9][0-9]-[0-9][0-9]-[0-9][0-9]-[0-9][0-9]-[0-9][0-9] | grep -v ' ' | sed -e 's%^/mnt/data%%g' | tr '\n' ' '`"
+  FND_ISOFILE=""
+  [ -n "$fromiso" ] && FND_ISOFILE="`find /mnt/data -maxdepth 2 -xdev -type f -name $(basename $fromiso) | grep -v ' ' | sed -e 's%^/mnt/data%%g' | tr '\n' ' '`"
+  if [ -n "$FND_ISOFILE" -a -z "$FND_PUPXXXSFS" ]; then
+      FND_PUPXXXSFS="/${DISTRO_FILE_PREFIX}-${DISTRO_VERSION}.sfs"
+  fi
   [ "$FND_MULTIFOLDER" != "" ] && echo "$ONEDEV $ONEFS $FND_MULTIFOLDER" >> /tmp/MULTIFOLDERS
   #all written on one line...
   echo -n "$ONEDEV $ONEFS " >> /tmp/PUPPYFILES
!  echo "$FND_QUIRKYKERNEL $FND_INITRDGZ $FND_PUPXXXSFS $FND_ZDRVSFS $FND_PUPSAVE2FS $FND_MULTIFOLDER $FND_FULLINSTALL $FND_SWAPFILEPUP $FND_ISOFILE" >> /tmp/PUPPYFILES
   [ -f /mnt/data/SAVEMARK ] && PSAVEMARK="`cat /mnt/data/SAVEMARK`" #v3.97 partition# that has or will-have ${DISTRO_FILE_PREFIX}save.
   #100318 fix any improper shutdowns... fsckme.err format: sda7,ext3,[/pupsave.2fs] (see rc.shutdown)
   FSCKME=""
***************
*** 543,548 ****
--- 548,557 ----
   fi
  fi
  
+ if [ -n "$fromiso" ]; then
+  ISOFILEFS=$(grep "$(basename $fromiso)" /tmp/PUPPYFILES)
+ fi
+ 
  if [ ! "$PDEV1" ];then
   #note, if 'psubdir' defined, PDEV1 already found above.
   #if only one line in /tmp/PUPPYFILES has initrd.gz then that must be the boot partition...
***************
*** 1148,1156 ****
    PPATTERN="/dev/$PUPSFSDEV "
    PUPSFSDEVMNTPT="`mount | grep "$PPATTERN" | cut -f 3 -d ' '`"
    if [ "$PUPSFSDEVMNTPT" = "" ];then
!    mntfunc $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro2 #-t $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro2
     PUPSFSDEVMNTPT="/mnt/dev_ro2"
     UMOUNTME="/mnt/dev_ro2" #mark for unmounting.
    fi
    
    if [ "$COPYPUPSFS2DIR" ];then #v406 copy ${DISTRO_FILE_PREFIX}-xxx.sfs to same place as ${DISTRO_FILE_PREFIX}save...
--- 1157,1176 ----
    PPATTERN="/dev/$PUPSFSDEV "
    PUPSFSDEVMNTPT="`mount | grep "$PPATTERN" | cut -f 3 -d ' '`"
    if [ "$PUPSFSDEVMNTPT" = "" ];then
!    if [ -n "$ISOFILEFS" ]; then
!        echo "Mounting /dev/${PUPSFSDEV}/${fromiso}"
!        mkdir -p /mnt/dev_ro3
!        mntfunc $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro3
!        LODEV="/dev/loop5"
!        losetup $LODEV /mnt/dev_ro3/${fromiso}
!        mntfunc iso9660 $LODEV /mnt/dev_ro2 #-t $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro2
!    else
!        mntfunc $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro2 #-t $PUPSFSFS /dev/$PUPSFSDEV /mnt/dev_ro2
!    fi
     PUPSFSDEVMNTPT="/mnt/dev_ro2"
     UMOUNTME="/mnt/dev_ro2" #mark for unmounting.
+   else
+       echo "$PUPSFSDEV is already mounted."
    fi
    
    if [ "$COPYPUPSFS2DIR" ];then #v406 copy ${DISTRO_FILE_PREFIX}-xxx.sfs to same place as ${DISTRO_FILE_PREFIX}save...
***************
*** 1181,1187 ****
      mv -f ${PUPSFSDEVMNTPT}${PUPSFSFILE} /mnt/tmpfs/
    else
     echo -e -n " \\033[1;35m${COPYMSG}\\033[0;39m" > /dev/console #purple.
!    cp -af ${PUPSFSDEVMNTPT}${PUPSFSFILE} /mnt/tmpfs/
    fi
    sync
    SFSBASENAME="`basename $PUPSFSFILE`"
--- 1201,1207 ----
      mv -f ${PUPSFSDEVMNTPT}${PUPSFSFILE} /mnt/tmpfs/
    else
     echo -e -n " \\033[1;35m${COPYMSG}\\033[0;39m" > /dev/console #purple.
!    cp -af ${PUPSFSDEVMNTPT}${PUPSFSFILE} /mnt/tmpfs/ ||  exec /bin/sh >/dev/console 2>&1
    fi
    sync
    SFSBASENAME="`basename $PUPSFSFILE`"
